frontend/src/components/core/Editor.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { ComponentProps } from 'react';
+import MonacoEditor from 'react-monaco-editor';
+
+type EditorProps = ComponentProps<typeof MonacoEditor>
```

Comment:
There's not a lot of information on `ComponentProps` -- it's not extremely commonly used since it doesn't work in class components -- but from what I understand it is essentially a way to create props without having to import a props interface/type.

Reply:
> I don't think there's a way to import the type of the Monaco Editor other than creating an instance of it, so typeof is necessary.

I don't believe this is true? I tried `MonacoEditorProps`, and this worked. For reference, on the [page you provided](https://www.npmjs.com/package/@types/react-monaco-editor), it says the following:

> react-monaco-editor provides its own type definitions, so you don't need @types/react-monaco-editor installed!

Please correct me if I'm wrong, but I think types are provided.

Reply:
Thanks for catching that! I didn't know that `MonacoEditorProps` existed. It works the same, so I'll replace that.